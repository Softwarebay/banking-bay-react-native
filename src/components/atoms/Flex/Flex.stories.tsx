import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Flex } from './Flex';

storiesOf('Atoms/Flex', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('Column', () => (
		<Flex direction="column">
			<View
				style={{ height: 100, width: 100, borderWidth: 2, marginBottom: 5 }}
			/>
			<View
				style={{ height: 100, width: 100, borderWidth: 2, marginBottom: 5 }}
			/>
		</Flex>
	))
	.add('Row', () => (
		<Flex direction="row">
			<View
				style={{ height: 100, width: 100, borderWidth: 2, marginLeft: 5 }}
			/>
			<View
				style={{ height: 100, width: 100, borderWidth: 2, marginLeft: 5 }}
			/>
		</Flex>
	));
