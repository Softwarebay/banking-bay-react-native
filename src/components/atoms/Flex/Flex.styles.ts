import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	flex: {
		alignItems: 'center',
	},
});
