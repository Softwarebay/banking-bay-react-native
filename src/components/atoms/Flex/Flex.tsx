import { StyleProp, View, ViewStyle } from 'react-native';

import { styles } from './Flex.styles';

interface Props {
	direction: 'column' | 'row';
	children: React.ReactNode;
	style?: StyleProp<ViewStyle>;
	testID?: string;
}

export const Flex = ({ direction, children, style, testID }: Props) => {
	return (
		<View
			style={[styles.flex, { flexDirection: direction }, style]}
			testID={testID}
		>
			{children}
		</View>
	);
};
