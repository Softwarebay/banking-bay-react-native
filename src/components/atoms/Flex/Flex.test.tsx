import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { View } from 'react-native';
import { Flex } from './Flex';

import type { Props } from './Flex';

describe('Flex', () => {
	const props: Omit<Props, 'direction'> = {
		children: (
			<>
				<View />
				<View />
			</>
		),
		testID: 'flex-test-id',
		style: {
			width: 100,
		},
	};

	const rowProps: Props = { direction: 'row', ...props };
	const columnProps: Props = { direction: 'column', ...props };

	it('should match snapshot - column', () => {
		const tree = testRenderer.create(<Flex {...columnProps} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - row', () => {
		const tree = testRenderer.create(<Flex {...rowProps} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render children in a column', () => {
		render(<Flex {...columnProps} />);

		const flex = screen.getByTestId('flex-test-id');

		expect(flex).toHaveStyle({ flexDirection: 'column' });
		expect(flex.children).toHaveLength(2);
	});

	it('should render children in a row', () => {
		render(<Flex {...rowProps} />);

		const flex = screen.getByTestId('flex-test-id');

		expect(flex).toHaveStyle({ flexDirection: 'row' });
		expect(flex.children).toHaveLength(2);
	});

	it('should apply styles from props', () => {
		render(<Flex {...rowProps} />);

		const flex = screen.getByTestId('flex-test-id');

		expect(flex).toHaveStyle({ width: 100 });
	});
});
