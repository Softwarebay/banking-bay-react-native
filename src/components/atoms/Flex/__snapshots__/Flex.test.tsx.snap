// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should match snapshot - column 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="flex-test-id"
>
  <View />
  <View />
</View>
`;

exports[`Flex should match snapshot - row 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="flex-test-id"
>
  <View />
  <View />
</View>
`;
